legend-engine-xt-persistence-test-runner/src/main/java/org/finos/legend/engine/testable/persistence/mapper/v2/MappingVisitors.java

```diff
@@ -0,0 +1,307 @@
+    {
+
+        @Override
+        public MergeStrategy visitNoActionIndicator(NoActionIndicator val)
+        {
+            return org.finos.legend.engine.persistence.components.ingestmode.merge.NoDeletesMergeStrategy.builder().build();
+        }
+
+        @Override
+        public MergeStrategy visitDeleteIndicator(DeleteIndicator val)
+        {
+            if (val instanceof DeleteIndicatorForTds)
+            {
+                return DeleteIndicatorMergeStrategy.builder()
+                        .deleteField(((DeleteIndicatorForTds) val).deleteField)
+                        .addAllDeleteValues(val.deleteValues)
+                        .build();
+            }
+            //todo (rengam) : Delete field for Graph fetch?
+            return DeleteIndicatorMergeStrategy.builder()
+                    //.deleteField(((DeleteIndicatorForGraphFetch) val).deleteFieldPath)
+                    .addAllDeleteValues(val.deleteValues)
+                    .build();
+        }
+    };
+
+    public static final ProcessingDimensionVisitor<TransactionMilestoning> MAP_TO_COMPONENT_PROCESSING_DIMENSION = new ProcessingDimensionVisitor<TransactionMilestoning>()
+    {
+
+        @Override
+        public TransactionMilestoning visitBatchId(org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.relational.temporality.processing.BatchId val)
+        {
+            return BatchId.of(
+                    val.batchIdIn == null || val.batchIdIn.isEmpty() ? BATCH_ID_IN_FIELD_DEFAULT : val.batchIdIn,
+                    val.batchIdOut == null || val.batchIdOut.isEmpty() ? BATCH_ID_OUT_FIELD_DEFAULT : val.batchIdOut);
+        }
+
+        @Override
+        public TransactionMilestoning visitDateTime(ProcessingDateTime val)
+        {
+            return TransactionDateTime.of(
+                    val.timeIn == null || val.timeIn.isEmpty() ? BATCH_TIME_IN_FIELD_DEFAULT : val.timeIn,
+                    val.timeOut == null || val.timeOut.isEmpty() ? BATCH_TIME_OUT_FIELD_DEFAULT : val.timeOut);
+        }
+
+        @Override
+        public TransactionMilestoning visitBatchIdAndDateTime(org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.relational.temporality.processing.BatchIdAndDateTime val)
+        {
+            return BatchIdAndDateTime.of(
+                    val.batchIdIn == null || val.batchIdIn.isEmpty() ? BATCH_ID_IN_FIELD_DEFAULT : val.batchIdIn,
+                    val.batchIdOut == null || val.batchIdOut.isEmpty() ? BATCH_ID_OUT_FIELD_DEFAULT : val.batchIdOut,
+                    val.timeIn == null || val.timeIn.isEmpty() ? BATCH_TIME_IN_FIELD_DEFAULT : val.timeIn,
+                    val.timeOut == null || val.timeOut.isEmpty() ? BATCH_TIME_OUT_FIELD_DEFAULT : val.timeOut);
+        }
+    };
+
+    public static final SourceDerivedDimensionVisitor<ValidityMilestoning> MAP_TO_COMPONENT_SOURCE_DERIVED_DIMENSION = new SourceDerivedDimensionVisitor<ValidityMilestoning>()
```

Comment:
I may have missed something here - is there a specific reason why we do a null check for processing dimension fields (batch_id_in etc.) but not so for source-derived dimension?
If we intend to make it parallel, I don't really think we need to do a null check here? because if those fields were null, even parser would have complained

Reply:
This was an exact copy of v1 - I've removed them.